### ISSUE TITLE
graphqurl 1.0.3 reported older punycode library is used and forcing to use older NodeJS versions

### ISSUE THREAD

**Issue by Kashish3009** [hasura/graphqurl#150]
### Problem Description:
graphqurl 1.0.3 and reports that punycode library used here is outdated and it forces them to use Node 19 (older version) where someone is using later versions of NodeJs

### Proposal:
To update the repository to use latest Punycode library.

### Output:

Output of `yarn why punycode -R`

graphqurl@npm:1.0.3 (via npm:^1.0.3)
   │  ├─ graphql-language-service-interface@npm:2.10.2 [36882] (via npm:^2.8.2 [36882])
   │  │  ├─ graphql-config@npm:4.5.0 [d912b] (via npm:^4.1.0 [d912b])
   │  │  │  └─ @graphql-tools/url-loader@npm:7.17.18 [422a5] (via npm:^7.9.7 [422a5])
   │  │  │     ├─ @whatwg-node/fetch@npm:0.8.8 (via npm:^0.8.0)
   │  │  │     │  └─ @whatwg-node/node-fetch@npm:0.3.6 (via npm:^0.3.6)
   │  │  │     │     └─ fast-url-parser@npm:1.1.3 (via npm:^1.1.3)
   │  │  │     │        └─ punycode@npm:1.4.1 (via npm:^1.3.2)

**Comment by manasag** [comment]
Thanks for opening the issue. Looking into this.

**Comment by manasag** [comment]
I have spent some time looking into this issue. This requires a lot of work at this stage to make this fully work with latest nodejs. The primary blocker is coming from the embedded GraphiQL UI (which runs via `gql endpoint -i`). It's is a very old modified version of GraphiQL, and the only way to make that work is to replace with newer versions of GraphiQL with similar modifications. This is a large undertaking.
If we remove the GraphiQL feature and focus only on the CLI and library, then its much smaller codebase to upgrade. I am going to create a new branch of `graphqurl` with GraphiQL feature disabled and tag it with a different version. This should help users who are ONLY interested in `graphqurl` as CLI for query execution on terminal and a graphql client. I should be able to release this in next couple of days.

**Comment by manasag** [comment]
Ok, so I am able to release 2.0.0 as described above. Here is the [PR](https://github.com/hasura/graphqurl/pull/152) for reference.

**Comment by spr-mweber3** [comment]
@manasag Sounds reasonable! Would work for me. Thank you!

**Comment by WTPOptAxe** [comment]
Will this new branch receive the same upgrades as `main` for things that don't impact GraphiQL in the future?

**Comment by manasag** [comment]
Yes, that will be the goal. I'll document this in the contribution guide. I would also like to mention that this is not actively maintained for feature requests, and will mostly only receive security updates on need basis. Feel free to contribute to this project.

Closing this issue. version [2.0.0](https://www.npmjs.com/package/graphqurl/v/2.0.0) has the fix. Branch https://github.com/hasura/graphqurl/tree/without-graphiql will be continued for this version minor changes.

**Comment by manasag** [comment]
Update: `main` now has the latest changes, which means from now on `graphqurl` do not have a GraphiQL support. the [v1.0.3](https://github.com/hasura/graphqurl/tree/v1.0.3) tag can be used to refer to the last commit with GraphiQL support. I have created this [issue](https://github.com/hasura/graphqurl/issues/153) to track bringing back support of GraphiQL.

Also to note that `graphqurl` is not actively maintained for feature requests. We will try to bring security and critical updates as needed. Contributions are welcomed.

Closing this issue. version [2.0.0](https://www.npmjs.com/package/graphqurl/v/2.0.0) is released with the fix.
